### PR TITLE
Handle removed machine labelsets differently

### PR DESCRIPTION
**Pull request by sudeephb**
Labelsets to be removed are found out by comparing current registry contents with current collected machine data. 
Fixes issue: #26

### CODE REVIEW

**Review by jneo8 — APPROVED** [review]
Approve only for seeing the functional test result

**Review by sudeephb — COMMENTED** [review]
recheck

File: tests/unit/test_exporter.py [hunk 1]
```diff
@@ -23,6 +23,46 @@ def test_parse_config(self, exporter_daemon):
...
+        for sample in samples:
+            sample.name = "example_gauge"
+        for metric in metrics:
+            metric.samples = samples
+        monkeypatch.setattr(
+            "prometheus_juju_exporter.exporter.CollectorRegistry.collect",
+            lambda r: metrics,
+        )
+        statsd.update_registry(stats)
+        assert len(statsd.metrics["example_gauge"].call_labels) == 2
```
Comment (rgildein) [inline tests/unit/test_exporter.py:64]: not-blocking: maybe we could be more specific here

Comment (sudeephb) [reply tests/unit/test_exporter.py]: I've updated it

**Review by sudeephb — COMMENTED** [review]
recheck

**Review by sudeephb — COMMENTED** [review]
recheck

**Pull request merged**